### PR TITLE
(firestore) F/E Validation Tests

### PR DESCRIPTION
**Pull request by kirstywilliams**
* F/E Validation Tests

### CODE REVIEW

File: packages/cloud_firestore/cloud_firestore/test/query_test.dart [hunk 1]
```diff
@@ -36,27 +35,32 @@ void main() {
...
+      expect(() => query.limitToLast(-1), throwsAssertionError);
     });
 
-    group('where()', () {
+    group('.where()', () {
       test('throws if field is invalid', () {
         expect(() => query.where(123), throwsAssertionError);
       });
 
+      // TODO(kirstywilliams): re-enable when decision made on skipping null values
```
Comment (Ehesp) [inline packages/cloud_firestore/cloud_firestore/test/query_test.dart:48]: So currently if the user passes in `null`, it skips the filter which I think is ok if we document it. The only one I'm not sure about is the `double.nan` - does native complain about it?

File: packages/cloud_firestore/cloud_firestore/test/write_batch_test.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          messagingSenderId: '123',
+          projectId: '123',
+        ));
+
+    firestore = Firestore.instance;
+    firestoreSecondary = Firestore.instanceFor(app: secondaryApp);
+  });
+
+  group("$WriteBatch", () {
+    test('may contain indirectly nested arrays', () {
```
Comment (Ehesp) [inline packages/cloud_firestore/cloud_firestore/test/write_batch_test.dart:56]: Not 100% sure what this is testing for?

File: packages/cloud_firestore/cloud_firestore/test/query_test.dart (hunk 1 above)
Comment (kirstywilliams) [reply packages/cloud_firestore/cloud_firestore/test/query_test.dart:48]: Ok cool, will remove this then. Not sure about `nan`, assuming it'll be the same as `null` - Not ran any native tests covering this.

File: packages/cloud_firestore/cloud_firestore/test/write_batch_test.dart (hunk 2 above)
Comment (kirstywilliams) [reply packages/cloud_firestore/cloud_firestore/test/write_batch_test.dart:56]: will remove this also, iirc checks any validation against the data object doesn't affect indirectly nested arrays.

**Pull request merged**